### PR TITLE
Chained equals feature [User Story #49]

### PR DESCRIPTION
**Pull request by 0matleb2**
Implements: **User Story** #49 

### WHAT kind of change does this PR introduce?

This pull request adds a feature that allows for chained equals presses to repeat the previous operation on the current result.

chainedequalsdemosmall

### HOW is this accomplished?
 * New fields in `CalculatorImpl`: `lastOperation`, `lastOperand`
 * Modified `handleEquals()` and `handleOperation()` in `CalculatorImpl`
 * Refactored duplicate code into new function `executeCalculation()` in `CalculatorImpl` 
 * Modified `complexTest()` to include chained equals
 * Added new tests to cover chained operations and chained equals

### Checklist
- [ ] Docs have been added / updated to reflect the changes
- [x] I have reviewed and tested the changes :white_check_mark:
- [x] I have squashed my commits to have a reasonable amount of commits
- [x] All of the code I have written adhere to the coding conventions outlined in the coding conventions section of the wiki

### CODE REVIEW

File: commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt [hunk 1]
```diff
@@ -58,21 +67,31 @@ class CalculatorImpl(calculator: Calculator) {
...
-            setValue(Formatter.doubleToString(firstNumber))
-            setFormula(operation.getFormula())
-            lastIsOperation = false
+        val operation : Operation?
+        if (operator != "") {   // Handle new operation
+            operation = OperationFactory.forId(operator!!, firstNumber, secondNumber)
+
+            if ((operation != null) && (digits > 0 || operation is UnaryOperation)) {
+                if (operation is UnaryOperation) {
+
```
Comment (jusleg) [inline commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt]: why not check if operation is not a unary operation (remove the blank if and just evaluate the else case)

Comment (0matleb2) [reply commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt]: Will fix this. I didn't have internet at the time and couldn't remember the negation syntax for kotlin.

File: commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt [hunk 2]
```diff
@@ -58,21 +67,31 @@ class CalculatorImpl(calculator: Calculator) {
...
-        val operation = OperationFactory.forId(operator!!, firstNumber, secondNumber)
-
-        if (operation != null && (digits > 0 || operation is UnaryOperation)) {
-            setAllClear()
-            resetValues()
-            firstNumber = operation.getResult()
-            setValue(Formatter.doubleToString(firstNumber))
-            setFormula(operation.getFormula())
-            lastIsOperation = false
+        val operation : Operation?
```
Comment (almiche) [inline commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt]: Why does this variable need to be immutable ?

File: commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt (hunk 1 above)
Comment (almiche) [reply commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt]: Agreed

File: commons/src/test/java/com/simpletools/calculator/commons/CalculatorImplTest.kt [hunk 3]
```diff
@@ -126,6 +126,72 @@ class CalculatorImplTest {
...
+        assertEquals("16", activity.getResult())
+        checkFormula("√256")
+    }
+
+    @Test
+    fun chainedEqualsBinaryOperationTest() {
+        calc.addDigit(10)
+        calc.handleOperation("plus")
+        calc.addDigit(1)
+        calc.handleEquals()     //11
```
Comment (almiche) [inline commons/src/test/java/com/simpletools/calculator/commons/CalculatorImplTest.kt]: Are these comments really necessary at this point ?

File: commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt (hunk 2 above)
Comment (jusleg) [reply commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt]: it's a const within the function. doesn't need to be `var`

File: commons/src/test/java/com/simpletools/calculator/commons/CalculatorImplTest.kt (hunk 3 above)
Comment (jusleg) [reply commons/src/test/java/com/simpletools/calculator/commons/CalculatorImplTest.kt]: I mean it's a lot of comments but it doesn't hurt

File: commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt (hunk 2 above)
Comment (0matleb2) [reply commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt]: This is a safe call operator. 'OperationFactory.forId()' return a safe Operation so this variable also needs to be declared safe.

File: commons/src/test/java/com/simpletools/calculator/commons/CalculatorImplTest.kt (hunk 3 above)
Comment (0matleb2) [reply commons/src/test/java/com/simpletools/calculator/commons/CalculatorImplTest.kt]: I find it helps to understand what the test is expected to do at a glance and without running the test.

**Comment by almiche** [comment]
✅ Styling is adequate and follows coding conventions outlined by the [Kotlin Foundation](https://kotlinlang.org/docs/reference/coding-conventions.html)
  * Styling is good

✅| New methods and functionalities have appropriate tests (testing edge cases + good coverage)
  * Everything has been thoroughly tested

✅| Branch has an acceptable amount of commits with descriptive names and
* Commits were properly squashed into one

✅| PR description matches the code outlined a has mentions to issues
  * Everything is up to spec. A gif is even included to show off the feature.

✅| This PR attempts to implement a specific feature (no code blob or epic pr)

Status: **Approved**
Comments:
